public/app/models/query/Condition.js

```diff
@@ -0,0 +1,84 @@
+// Based on http://kindohm.github.io/knockout-query-builder/js/condition.js
+(function () {
+    'use strict';
+
+    define(
+        [
+            'lodash',
+            'knockout'
+        ],
+        function (_, ko) {
+            return function (queryBuilder) {
+                this.selectedField = ko.observable();
+                this.selectedComparator = ko.observable();
+                this.value = ko.observable('');
+
+                this.fields = ko.pureComputed(function () {
+                    return queryBuilder.fields();
+                });
+
+                this.comparators = ko.pureComputed(function () {
+                    return [
+                        {
+                            key: 'contains',
+                            label: 'contains',
+                            queryFormat: function (value) {
+                                return '"' + value + '"';
+                            }
+                        },
+                        {
+                            key: '!contains',
+                            label: 'does not contain',
+                            queryFormat: function (value) {
+                                return '-"' + value + '"';
+                            }
+                        },
+                        {
+                            key: 'starts',
+                            label: 'starts with',
+                            queryFormat: function (value) {
+                                return '"' + value + '"*';
+                            }
+                        },
+                        {
+                            key: '!starts',
+                            label: 'does not start with',
+                            queryFormat: function (value) {
+                                return '-"' + value + '"*';
+                            }
+                        }
+                    ];
+                });
+
+                this.query = ko.pureComputed(function () {
+                    var field, comparator, value;
+                    field = this.selectedField();
+                    comparator = this.selectedComparator();
+                    value = this.value();
+                    return (field && comparator && value) ?
+                        {
+                            field: field,
+                            comparator: comparator,
+                            value: value
+                        } :
+                        undefined;
+                }.bind(this));
+
+                this.parse = function (query) {
```

Comment:
Inverse of `query` - takes the query object as passed to the `searchService` (which is brought in from the URL) and parses it into the observables in this object.